src/roi/computeRois.ts

```diff
@@ -64,24 +76,65 @@ export function computeRois(roiMapManager: RoiMapManager): void {
   roiMapManager.whiteRois = new Array<Roi>(map.nbPositive);
   roiMapManager.blackRois = new Array<Roi>(map.nbNegative);
 
+  for (let i = 0; i < map.nbNegative; i++) {
+    let blackRoi = new Roi(map, -i);
+    blackRoi.origin = { row: blacks[i].minRow, column: blacks[i].minColumn };
+    blackRoi.width = blacks[i].maxColumn - blacks[i].minColumn + 1;
+    blackRoi.height = blacks[i].maxRow - blacks[i].minRow + 1;
+    blackRoi.surface = blacks[i].surface;
+    blackRoi.id = blacks[i].id;
+    //computes minX,minY,maxX,maxY for black surfaces
+    for (let y = 0; y < map.height; y++) {
+      for (let x = 0; x < map.width; x++) {
+        let target = y * map.width + x;
+        if (map.data[target] === blackRoi.id) {
+          const roi = blackRoi;
+          if (x < roi.minX) {
+            roi.minX = x;
+          }
+          if (x > roi.maxX) {
+            roi.maxX = x;
+          }
+          if (y < roi.minY) {
+            roi.minY = y;
+          }
+          if (y > roi.maxY) {
+            roi.maxY = y;
+          }
+        }
+      }
+    }
```

Comment:
I'm convince maxX, maxY, minX, minY are useless and already in origin, width, height.